slatedb/src/compactor.rs

```diff
@@ -595,30 +563,18 @@ impl CompactorEventHandler {
     }
 
     #[instrument(level = "debug", skip_all, fields(id = tracing::field::Empty))]
-    async fn submit_compaction(&mut self, compactor_job: CompactorJob) -> Result<(), SlateDBError> {
-        let request = compactor_job.compactor_job_request();
+    async fn submit_compaction(&mut self, job: CompactorJob) -> Result<(), SlateDBError> {
         // Validate the candidate compaction; skip invalid ones
-        if let Err(e) = self.validate_compaction(request) {
+        if let Err(e) = self.validate_compaction(job.spec()) {
             warn!("invalid compaction [error={:?}]", e);
             return Ok(());
         }
 
-        self.state.submit_compactor_job(compactor_job.clone());
-        // Generate a compactionJob id
-        let id = self.rand.rng().gen_ulid(self.system_clock.as_ref());
-        tracing::Span::current().record("id", tracing::field::display(&id));
-        let result = self.state.submit_compaction(id, compactor_job.clone());
-        match result {
-            Ok(_) => {
-                // TODO(sujeetsawala): Add compaction plan to object store with Pending status
-                self.start_compaction(id, compactor_job).await?;
-            }
-            Err(err) => {
-                // TODO(sujeetsawala): Add compaction plan to object store with Failed status
-                self.state.remove_compactor_job(&compactor_job.id());
-                warn!("invalid compaction [error={:?}]", err);
-            }
-        }
+        self.state.add_job(job.clone())?;
```

Comment:
Since the job is becoming more of an "entity" in our model (in the sense that it is our in-memory representation of an actual "thing" that exists in the system (vs just a description of it)) I don't really love that we're cloning it. We should just create it, register it with the state and then access it by reference. So I think something like:
```
let job_id = job.jd();
let job: &CompactorJob = self.state.add_job(job)?;
self.start_compaction(job).await?; // construct the attempt here and return its metadata so it can be added to the job when we want to track attempts/progress
```

I'd be ok with punting this though since it's net-new. I can take it up after this patch